hardware_interface/CMakeLists.txt

```
@@ -123,65 +81,73 @@ if(BUILD_TESTING)
   ament_target_dependencies(test_joint_handle rcpputils)
 
   ament_add_gmock(test_component_interfaces test/test_component_interfaces.cpp)
-  target_link_libraries(test_component_interfaces ${PROJECT_NAME})
+  target_link_libraries(test_component_interfaces hardware_interface)
 
   ament_add_gmock(test_component_parser test/test_component_parser.cpp)
-  target_link_libraries(test_component_parser ${PROJECT_NAME})
+  target_link_libraries(test_component_parser hardware_interface)
   ament_target_dependencies(test_component_parser ros2_control_test_assets)
 
   add_library(test_components SHARED
     test/test_components/test_actuator.cpp
     test/test_components/test_sensor.cpp
     test/test_components/test_system.cpp)
-  target_link_libraries(test_components ${PROJECT_NAME})
+  target_link_libraries(test_components hardware_interface)
   ament_target_dependencies(test_components
     pluginlib)
   install(TARGETS test_components
     DESTINATION lib
   )
   pluginlib_export_plugin_description_file(
-    ${PROJECT_NAME} test/test_components/test_components.xml)
+    hardware_interface test/test_components/test_components.xml)
 
   add_library(test_hardware_components SHARED
   test/test_hardware_components/test_single_joint_actuator.cpp
   test/test_hardware_components/test_force_torque_sensor.cpp
   test/test_hardware_components/test_two_joint_system.cpp
   test/test_hardware_components/test_system_with_command_modes.cpp
   )
-  target_link_libraries(test_hardware_components ${PROJECT_NAME})
+  target_link_libraries(test_hardware_components hardware_interface)
   ament_target_dependencies(test_hardware_components
     pluginlib)
   install(TARGETS test_hardware_components
     DESTINATION lib
   )
   pluginlib_export_plugin_description_file(
-    ${PROJECT_NAME} test/test_hardware_components/test_hardware_components.xml
+    hardware_interface test/test_hardware_components/test_hardware_components.xml
   )
 
   ament_add_gmock(test_resource_manager test/test_resource_manager.cpp)
-  target_link_libraries(test_resource_manager ${PROJECT_NAME})
+  target_link_libraries(test_resource_manager hardware_interface)
   ament_target_dependencies(test_resource_manager ros2_control_test_assets)
 
   ament_add_gmock(test_generic_system test/mock_components/test_generic_system.cpp)
   target_include_directories(test_generic_system PRIVATE include)
-  target_link_libraries(test_generic_system ${PROJECT_NAME})
+  target_link_libraries(test_generic_system hardware_interface)
   ament_target_dependencies(test_generic_system
     pluginlib
     ros2_control_test_assets
   )
 endif()
 
-ament_export_include_directories(
-  include
+install(
+  DIRECTORY include/
+  DESTINATION include/hardware_interface
 )
-ament_export_libraries(
-  fake_components
-  mock_components
-  ${PROJECT_NAME}
+install(
+  TARGETS
+    fake_components
+    mock_components
+    hardware_interface
+  EXPORT export_hardware_interface
```

Do we need here also the following?

```suggestion
  EXPORT export_fake_components
  EXPORT export_mock_components
  EXPORT export_hardware_interface
```

---

We are exploring them. The name here isn't important. They are all in the install target list.